easybuild/tools/toolchain/toolchain.py

```diff
@@ -1025,6 +1028,28 @@ def prepare_rpath_wrappers(self, rpath_filter_dirs=None, rpath_include_dirs=None
             else:
                 self.log.debug("Not installing RPATH wrapper for non-existing command '%s'", cmd)
 
+    def handle_sysroot(self):
+        """
+        Extra stuff to be done when alternate system root is specified via --sysroot EasyBuild configuration option.
+
+        * Update $PKG_CONFIG_PATH to include sysroot location to pkg-config files (*.pc).
+        """
+        sysroot = build_option('sysroot')
+        if sysroot:
+            # update $PKG_CONFIG_PATH to include sysroot location to pkg-config files (*.pc)
+            sysroot_pc_paths = [os.path.join(sysroot, 'usr', libdir, 'pkgconfig') for libdir in ['lib', 'lib64']]
```

Comment:
There is actually reasons to do this even when sysroot is not used.
I.e. picking up the system pkgconfig files as a last resort... It's really no different than when using sysroot.

Reply:
I think `pkg-config` will check the system paths by default, no?

Reply:
No, it will only check what's in PKG_CONFIG_PATH

Reply:
I'm not sure it's a good idea to always set `$PKG_CONFIG_PATH`, since generally we want to *avoid* that stuff is picked up from the OS? Unless `--filter-deps` is used, maybe...

You can always just set `$PKG_CONFIG_PATH` outside of `eb` to make sure it `*.pc` files are picked up from the OS (since we're careful here no to overwrite existing paths in `$PKG_CONFIG_PATH`)...

Reply:
True, but why then set it to sysroot in the first place??

Reply:
Because using `--sysroot` (in combination with `--filter-deps`) should imply that `$PKG_CONFIG_PATH` is set to the same path.

Reply:
With filter-deps perhaps but this doesn't check filter-deps being set at all.

And sysroot set or not isn't really relevant. It should behave the same in both cases.

So, I'd vote for not setting it at all. Set it from the outside if you need it.

Reply:
Perhaps a config option to decide if PKG_CONFIG gets set or not when filter-deps is used??
But that's a bit too specific for my taste.

Reply:
Yeah, that's very specific indeed...

To me it makes sense to do this automatically when `--sysroot` is used, since then you're clearly building on top of something else where you want to use libraries/tools from.

That's not the case when building without a custom sysroot?

Reply:
Well, that's one way to look at it.

Setting it only when filter-deps is in use would make more sense to me though, but I'm not going to argue that.

Should we merge as is then?